sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -1507,4 +1508,16 @@ class JDBCSuite extends QueryTest
       checkNotPushdown(sql("SELECT name, theid FROM predicateOption WHERE theid = 1")),
       Row("fred", 1) :: Nil)
   }
+
+  test("SPARK-26383 throw IllegalArgumentException if url is wrong") {
+    val e = intercept[IllegalArgumentException] {
+      val opts = Map(
+        "url" -> "jdbc:mysql://localhost/db",
```

Comment:
Do you want to use a more obviously invalid URL here? or does it only trigger this scenario if the URL is valid but can't be opened?

Reply:
The driver returns `null` if it realizes it is the wrong kind of driver to connect to the given URL. I used `mysql` url with `postgresql` driver.